src/components/common/textcopypopover/TextCopyPopover.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import { Popover, TextArea } from '@patternfly/react-core-updated';
+import { CopyIcon } from '@patternfly/react-icons';
+import { Instance, BasicPlacement } from 'tippy.js';
+
+import './textcopypopover.scss';
+
+interface TextCopyPopoverProps {
+  text: string,
+  headerContent: JSX.Element | string,
+  isVisible?: boolean,
+  position?: BasicPlacement,
+  subheaderContent?: string,
+  rows?: number,
+
+  tabIndex?: number,
+  className?: string
+  shouldClose?: (instance: Instance) => void,
+  onMouseDown?: (event: React.MouseEvent) => void,
+  onBlur?: (event: React.FocusEvent) => void,
+  maxWidth?: string,
+}
+
+interface TextCopyPopoverState {
+  copied: boolean
+}
+
+class TextCopyPopover extends React.Component<TextCopyPopoverProps, TextCopyPopoverState> {
```

Comment:
@matankb can you use <ClipboardCopy> in patternfly here? I think it would be helpful

Reply:
done!